cds_books/patrons/views.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio App ILS Patrons views."""
+
+from copy import deepcopy
+from functools import wraps
+
+from flask import Blueprint, abort, current_app, jsonify
+from invenio_app_ils.circulation.views import need_permissions
+from invenio_app_ils.search.api import PatronsSearch
+from invenio_circulation.search.api import search_by_patron_item_or_document
+from invenio_rest import ContentNegotiatedMethodView
+
+from .serializers import patron_loans_serializer
+
+
+def pass_patron_from_es():
+    """Decorator to retrieve a patron from ES."""
+    def pass_patron_decorator(f):
+        @wraps(f)
+        def inner(self, person_id, *args, **kwargs):
+            results = PatronsSearch().filter(
```

Comment:
Does Flask with `string:patron_id` validates already if you put some weird thing? Sanitize the param?

Reply:
I will check that!

Reply:
I made the converter to accept only integers for person_id. Flask says that if is not a valid integer it results to a 404! http://exploreflask.com/en/latest/views.html#built-in-converters